lib/wm_util.ml

```diff
@@ -111,11 +111,71 @@ module MediaType = struct
     loop ranges
 end
 
-module Date = struct
-  open CalendarLib
+module type Date_sig = sig
+  type t
+  val now : unit -> t
+  val parse_rfc1123_date_exn : string -> t
+  val parse_rfc1123_date : string -> t option
+end
+
+module type PtimeNow_sig = sig
+  val now : unit -> Ptime.t
+end
+
+(** MakePtimeDate creates a Ptime based Date module, that can be used with the
+    Webmachine.Make(IO)(Date) functor.
+
+    Examples:
+    module DateMock = Wm_util.MakePtimeDate(struct
+      let now = fun () -> Ptime.v (17591, 0L)
+    end)
+
+    module DatePtimeClock = Wm_util.MakePtimeDate(struct
+      let now = Ptime_clock.now
+    end)
+
+    module DateMirage = Wm_util.MakePtimeDate(struct
+      let now = fun () -> Ptime.v @@ Pclock.now_d_ps clock
+    end)
+*)
+module MakePtimeDate(NOW:PtimeNow_sig) : Date_sig = struct
+
+  type t = Ptime.t
+
+  let now = NOW.now
 
   let parse_rfc1123_date_exn s =
```

Comment:
Cool! I will have to look at the HTTP spec, but I think only `GMT` is allowed as a timezone. I guess there's no harm in supporting the others though?

Reply:
No, the others are also allowed, but deprecated.